config-linux.md

```diff
@@ -270,13 +270,16 @@ For more information, see the kernel cgroups documentation about [memory][cgroup
 **`memory`** (object, OPTIONAL) represents the cgroup subsystem `memory` and it's used to set limits on the container's memory usage.
 For more information, see the kernel cgroups documentation about [memory][cgroup-v1-memory].
 
-The following parameters can be specified to set up the controller:
+Values for memory specify the limit in bytes, or `-1` for unlimited memory.
+
+* **`limit`** *(int64, OPTIONAL)* - sets limit of memory usage
+* **`reservation`** *(int64, OPTIONAL)* - sets soft limit of memory usage
+* **`swap`** *(int64, OPTIONAL)* - sets limit of memory+Swap usage
+* **`kernel`** *(int64, OPTIONAL)* - sets hard limit for kernel memory
+* **`kernelTCP`** *(int64, OPTIONAL)* - sets hard limit for kernel TCP buffer memory
+
+For `swappiness` the values are from `0` (almost never swap) to `100` (very likely to be swapped).
```

Comment:
> If you believe the important thing is what is written to the `memory.swappiness` we should not be discussing this PR at all, we should change all these values to strings, which is what is actually written to the files.

Yeah, and a cgroup config like:

```
"<controller>": {
    "<filename-suffix>": "<string to write>"
}
```

For example:

```
"memory": {
  "swappiness": "60",
  "soft_limit_in_bytes": "-1",
}
```

would be as expressive as what we have now and a lot easier to document.  And it extends to other controllers and properties without needing any spec changes.

The counter-argument is that that approach degenerates into an array-of-syscalls config (e.g. if you [set capabilities as integers instead of `CAP_*` strings][1].  Drawing the line somewhere is useful to keep configs human-readable, but `"swappiness": 0` and `"swappiness": "0"` are equally legible to me.

But overhauling the cgroup-config API is a big change with a 1.0 freeze in place, so in the short term I'd rather just avoid baking more kernel-defined semantics into the spec in situations where the runtime implementation can ignore those semantics.

[1]: https://github.com/opencontainers/runtime-spec/pull/790#discussion_r115609459